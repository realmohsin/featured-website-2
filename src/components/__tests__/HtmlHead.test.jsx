import React from "react";
import { render } from "@testing-library/react";
import HtmlHead from "../HtmlHead";

// Mock Gatsby's useStaticQuery and graphql
jest.mock("gatsby", () => ({
  useStaticQuery: jest.fn(() => ({
    site: {
      siteMetadata: {
        title: "Romano Law PLLC",
        description: "Business, Entertainment & Dispute Resolution Law Firm",
        author: "Romano Law",
      },
    },
  })),
  graphql: jest.fn(),
}));

// Mock react-helmet
jest.mock("react-helmet", () => ({
  Helmet: ({ children }) => <div data-testid="helmet">{children}</div>,
}));

describe("HtmlHead Component", () => {
  it("renders with default site metadata", () => {
    const { container } = render(<HtmlHead />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders with custom title", () => {
    const { container } = render(<HtmlHead title="Custom Page Title" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders with custom description", () => {
    const { container } = render(
      <HtmlHead description="Custom page description for SEO" />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders with both custom title and description", () => {
    const { container } = render(
      <HtmlHead
        title="About Us - Romano Law"
        description="Learn about Romano Law PLLC and our experienced legal team"
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
