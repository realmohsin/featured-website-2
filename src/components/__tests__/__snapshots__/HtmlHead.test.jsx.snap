// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HtmlHead Component renders with both custom title and description 1`] = `
<div
  data-testid="helmet"
>
  <title>
    About Us - Romano Law
  </title>
  <meta
    content="Learn about Romano Law PLLC and our experienced legal team"
    name="description"
  />
  <link
    href="https://fonts.googleapis.com/css2?family=Noto+Sans+JP&family=Piazzolla&family=PT+Serif&family=Noto+Serif+JP&family=Cinzel&family=Playfair+Display&display=swap"
    rel="stylesheet"
  />
  <link
    href="https://fonts.gstatic.com"
    rel="preconnect"
  />
  <link
    href="https://fonts.googleapis.com/css2?family=Roboto&display=swap"
    rel="stylesheet"
  />
</div>
`;

exports[`HtmlHead Component renders with custom description 1`] = `
<div
  data-testid="helmet"
>
  <title>
    Romano Law PLLC
  </title>
  <meta
    content="Custom page description for SEO"
    name="description"
  />
  <link
    href="https://fonts.googleapis.com/css2?family=Noto+Sans+JP&family=Piazzolla&family=PT+Serif&family=Noto+Serif+JP&family=Cinzel&family=Playfair+Display&display=swap"
    rel="stylesheet"
  />
  <link
    href="https://fonts.gstatic.com"
    rel="preconnect"
  />
  <link
    href="https://fonts.googleapis.com/css2?family=Roboto&display=swap"
    rel="stylesheet"
  />
</div>
`;

exports[`HtmlHead Component renders with custom title 1`] = `
<div
  data-testid="helmet"
>
  <title>
    Custom Page Title
  </title>
  <meta
    content="Business, Entertainment & Dispute Resolution Law Firm"
    name="description"
  />
  <link
    href="https://fonts.googleapis.com/css2?family=Noto+Sans+JP&family=Piazzolla&family=PT+Serif&family=Noto+Serif+JP&family=Cinzel&family=Playfair+Display&display=swap"
    rel="stylesheet"
  />
  <link
    href="https://fonts.gstatic.com"
    rel="preconnect"
  />
  <link
    href="https://fonts.googleapis.com/css2?family=Roboto&display=swap"
    rel="stylesheet"
  />
</div>
`;

exports[`HtmlHead Component renders with default site metadata 1`] = `
<div
  data-testid="helmet"
>
  <title>
    Romano Law PLLC
  </title>
  <meta
    content="Business, Entertainment & Dispute Resolution Law Firm"
    name="description"
  />
  <link
    href="https://fonts.googleapis.com/css2?family=Noto+Sans+JP&family=Piazzolla&family=PT+Serif&family=Noto+Serif+JP&family=Cinzel&family=Playfair+Display&display=swap"
    rel="stylesheet"
  />
  <link
    href="https://fonts.gstatic.com"
    rel="preconnect"
  />
  <link
    href="https://fonts.googleapis.com/css2?family=Roboto&display=swap"
    rel="stylesheet"
  />
</div>
`;
